src/flux/store.js

```diff
@@ -414,14 +464,56 @@ class Store extends EventEmitter {
 
   importCustomYAML = (customYAML) => {
     logger.log("importCustomYAML - customYAML", customYAML);
-    this.initFlowChart(customYAML);
+    this.createNewFlow(customYAML);
     this.closeModal();
     this.emit("update-flowchart");
   };
 
-  loadExample = (exampleName) => {
-    const flow = exampleYAML[exampleName];
-    this.initFlowChart(flow);
+  loadFlow = (flowId) => {
+    _store.selectedFlow = flowId;
+    this.emit("update-flowchart");
+  };
+
+  updateFlow = (newFlow) => {
+    _store.flows[_store.selectedFlow].flow = newFlow;
+    this.emit("update-flowchart");
+  };
+
+  createNewFlow = (customYAML) => {
+    let prefixString = "Custom Flow";
+
+    let userFlows = Object.entries(_store.flows)
+      .filter(([id, flow]) => flow.name.startsWith(prefixString))
+      .map(([id, flow]) => flow);
+
+    const flowNumbers = userFlows
+      .map((f) => parseInt(f.name.substring(prefixString.length)) || 0)
+      .sort((a, b) => a - b);
+
+    const largestNumber = flowNumbers[flowNumbers.length - 1] || 0;
```

Comment:
what's the difference between all that code and just `const nFlows=  _store.flows.length`. And is `largestNumber` not just the count?

Reply:
The difference is flows object contains examples and potentially connected flows, so the count of total items does not match the number of user generated flows that use the default name

Reply:
Why don't you just filter for this and take the length?

Reply:
If we filter and take the length, an item can be removed from the middle of a sequence causing the highest number and the length of the array to fall out of line. Generating a name based on that condition would result in two items with the same name